### PR TITLE
Add caps lock warning to OsuPasswordTextBox

### PR DESCRIPTION
**Pull request by jorolf**

### CODE REVIEW

File: osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs [hunk 1]
```diff
@@ -15,6 +19,35 @@ public class OsuPasswordTextBox : OsuTextBox
...
+            });
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost host)
+        {
+            this.host = host;
+        }
+
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
```
Comment (MrTheMake) [inline osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs]: Shouldn't the warning be updated regardless of whether caps is pressed while this textbox is focused?

File: osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs [hunk 2]
```diff
@@ -15,6 +19,35 @@ public class OsuPasswordTextBox : OsuTextBox
...
+
+        public OsuPasswordTextBox()
+        {
+            Add(warning = new CapsWarning
+            {
+                Size = new Vector2(20),
+                Origin = Anchor.CentreRight,
+                Anchor = Anchor.CentreRight,
+                Margin = new MarginPadding { Right = 10 },
+                Alpha = 0,
```
Comment (MrTheMake) [inline osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs]: The initial alpha value should depend on the current caps state.

File: osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs]: Pretty sure it should start displaying when the `TextBox` receives focus, and hide after losing focus.

Comment (jorolf) [reply osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs]: _done_

**Review by smoogipoo — APPROVED** [review]
👍

**Pull request merged**